packages/core/src/telemetry/clearcut-logger/clearcut-logger.ts

```diff
@@ -413,6 +421,20 @@ export class ClearcutLogger {
     });
   }
 
+  getProxyAgent() {
+    const proxyUrl = this.config?.getProxy();
+    if (!proxyUrl) return undefined;
+    // undici which is widely used in the repo can only support http & https proxy protocol,
+    // https://github.com/nodejs/undici/issues/2224
+    // as @Dcatfly metioned, `HttpsProxyAgent` can support https & http proxy protocol at the same time,
```

Comment:
Can drop this line and the next given they're a bit more code reviewy for a comment